include/fdreadoutlibs/wibeth/tpg/ProcessAbsRSAVX2.hpp

```diff
@@ -129,15 +129,24 @@ process_window_rs_avx2(ProcessingInfo<NREGISTERS>& info)
 
       // Instead of using floats in the calcualation of the RS we multiply by 10 and 
       // do operations on the integers. In the end we divide by 10. 
-     
+    
+      /* 
       __m256i first_part = _mm256_mullo_epi16(RS, R_factor);
       __m256i second_part = _mm256_mullo_epi16(_mm256_abs_epi16(s), scale_factor);
       //RS = swtpg_wibeth::_mm256_div_epi16(_mm256_add_epi16(first_part, second_part), 10);
 
       __m256i first_part_div = _mm256_div_epi16(first_part, 10);
       __m256i second_part_div = _mm256_div_epi16(second_part, 10);
       RS = _mm256_add_epi16(first_part_div, second_part_div);
+      */
+
+      __m256i first_part = _mm256_div_epi16(RS, 10);
+      first_part =_mm256_mullo_epi16(first_part, R_factor); 
+
+      __m256i second_part = _mm256_div_epi16(_mm256_abs_epi16(s), 10);
+      second_part = _mm256_mullo_epi16(second_part, scale_factor);
 
+      RS = _mm256_add_epi16(first_part, second_part);
```

Comment:
We should be using saturated addition here. Otherwise, we can overflow into negative values.

```suggestion
      RS = _mm256_adds_epi16(first_part, second_part);
```